pkgs/development/compilers/cmdstan/default.nix

```diff
@@ -1,16 +1,30 @@
-{ stdenv, fetchurl, python, runtimeShell }:
+{ stdenv, fetchurl, python, runtimeShell, bash }:
 
-stdenv.mkDerivation {
-  name = "cmdstan-2.17.1";
+let
+  name = "cmdstan";
+  version = "2.25.0";
+in stdenv.mkDerivation {
+  name = "${name}-${version}";
 
   src = fetchurl {
-    url = "https://github.com/stan-dev/cmdstan/releases/download/v2.17.1/cmdstan-2.17.1.tar.gz";
-    sha256 = "1vq1cnrkvrvbfl40j6ajc60jdrjcxag1fi6kff5pqmadfdz9564j";
+    url =
```

Comment:
```
  src = fetchFromGitHub {
    owner = "stan-dev";
    repo = pname;
    rev = "v${version}";
    sha256 = "TODO";
  };
```

Reply:
stanc3's default.nix pins nixpkgs 19.09.
I haven't figured out a working stanc derivation on nixpkgs-unstable that doesn't use import-from-derivation (see below). 

Using the release tarball with `fetchTarball` includes the binaries from stanc3. Using `fetchFromGitHub` excludes the `bin/` directory so it doesn't work.

```nix
{ stdenv, fetchFromGitHub, python, runtimeShell, bash }:
let
  stanc3 = import (fetchTarball {
    url = "https://github.com/stan-dev/stanc3/archive/v2.25.0.tar.gz";
    sha256 = "sha256:01iak3ac2hzdrjbyfcsbqiizjd2b4jmb422fxxia0vcw799jwhay";
  });
in stdenv.mkDerivation rec {
  pname = "cmdstan";
  version = "2.25.0";

  src = fetchFromGitHub {
    owner = "stan-dev";
    repo = "cmdstan";
    rev = "v${version}";
    sha256 = "sha256:1s49rhak2phkzqfjx1xmj74frqi79p99n3kcmndm3fbq6b695ag1";
    fetchSubmodules = true;
  };

  buildFlags = [ "build" ];
  enableParallelBuilding = true;

  postPatch = ''

    substituteInPlace stan/lib/stan_math/make/libraries \
        --replace "/usr/bin/env bash" ${bash}/bin/bash

    substituteInPlace stan/lib/stan_math/lib/boost_1.72.0/bootstrap.sh \
        --replace '`which echo`' echo

    patchShebangs .

    mkdir bin
    cp -r ${stanc3}/bin/* bin
      '';

  # Hack to avoid TMPDIR in RPATHs.
  # Copied from https://github.com/NixOS/nixpkgs/commit/f5c568446a12dbf58836925c5487e5cdad1fa578
  preFixup = ''
    rm -rf "$(pwd)"
    mkdir "$(pwd)"
  '';

  doCheck = true;
  checkInputs = [ python ];
  checkPhase = "python ./runCmdStanTests.py src/test/interface"; # see #5368

  installPhase = ''
    mkdir -p $out/opt $out/bin
    cp -r . $out/opt/cmdstan
    ln -s $out/opt/cmdstan/bin/stanc $out/bin/stanc
    ln -s $out/opt/cmdstan/bin/stansummary $out/bin/stansummary
    cat > $out/bin/stan <<EOF
    #!${runtimeShell}
    make -C $out/opt/cmdstan "\$(realpath "\$1")"
    EOF
    chmod a+x $out/bin/stan
  '';

  meta = {
    description = "Command-line interface to Stan";
    longDescription = ''
      Stan is a probabilistic programming language implementing full Bayesian
      statistical inference with MCMC sampling (NUTS, HMC), approximate Bayesian
      inference with Variational inference (ADVI) and penalized maximum
      likelihood estimation with Optimization (L-BFGS).
    '';
    homepage = "https://mc-stan.org/interfaces/cmdstan.html";
    license = stdenv.lib.licenses.bsd3;
    platforms = stdenv.lib.platforms.all;
  };
}

```